python-venv/cmstools/test/cfs_sessions_rc_test/cfs/session.py

```diff
@@ -0,0 +1,259 @@
+
+
+def get_all_cfs_sessions_v3(cfs_session_name_contains: str, cfs_version: CFS_VERSIONS_STR, limit: int, retry: bool) -> MultiSessionsGetResult:
+    url = CFS_SESSIONS_URL_TEMPLATE.format(api_version=cfs_version)
+    params = get_cfs_sessions_list_params(
+        cfs_session_name_contains=cfs_session_name_contains,
+        cfs_version=cfs_version,
+        limit=limit
+    )
+    all_sessions: list[JsonDict] = []
+    after_id: str | None = None
+
+    try:
+        while True:
+            if after_id:
+                params["after_id"] = after_id
+            resp = make_request(url=url, params=params, retry=retry)
+            if resp.status_code != HTTP_OK:
+                logger.error("Unexpected return code %d from GET query to %s: %s", resp.status_code, url, resp.text)
+                return MultiSessionsGetResult(status_code=resp.status_code, error_message=resp.text)
+            data = resp.json()
+            sessions = get_session_data(data)
+            if not sessions:
+                break
+            all_sessions.extend(sessions)
+            next_id = get_next_id(data)
+            if not next_id:
+                break
+            after_id = next_id
+        logger.debug("Session data found: %s", all_sessions)
+        return MultiSessionsGetResult(status_code=HTTP_OK, session_data=all_sessions)
+    except requests.exceptions.Timeout:
+        logger.exception("Request timed out for GET query to %s", url)
+        return MultiSessionsGetResult(status_code=0, timed_out=True, error_message="Request timed out")
+    except Exception as exc:
+        logger.exception("Exception during CFS session get: %s", str(exc))
+        return MultiSessionsGetResult(status_code=0, error_message=str(exc))
+
+
+def get_cfs_sessions_list(cfs_session_name_contains: str, cfs_version: CFS_VERSIONS_STR, limit: int, retry: bool = True) -> MultiSessionsGetResult:
+    """
+    Returns a list of CFS sessions with the specified name prefix and pending status.
+    """
+    if cfs_version == "v2":
+        return get_all_cfs_sessions_v2(
+            cfs_session_name_contains=cfs_session_name_contains,
+            cfs_version=cfs_version, retry=retry)
+
+    return get_all_cfs_sessions_v3(
+        cfs_session_name_contains=cfs_session_name_contains,
+        cfs_version=cfs_version,
+        limit=limit, retry=retry)
+
+
+def get_expected_delete_sessions_status_codes(cfs_version: CFS_VERSIONS_STR) -> int:
+    """
+    Get expected status code for delete sessions based on CFS version.
+    """
+    if cfs_version == "v2":
+        return HTTP_NO_CONTENT
+    return HTTP_OK
+
+
+def delete_cfs_sessions(cfs_session_name_contains: str, cfs_version: CFS_VERSIONS_STR, limit: int) -> None:
+    """
+    Delete all CFS sessions with the specified name prefix and pending status.
+    """
+    result = get_cfs_sessions_list(cfs_session_name_contains, cfs_version, limit)
+    if result.status_code != HTTP_OK or not result.session_data:
```

Comment:
If the get request returns non-ok, that doesn't imply no sessions exist, right? I don't think the multi-get ever returns 404 or similar.

Reply:
Those changes don't address my concern, though. The way the code is currently written, if the get request returns 400, it is not interpreting that as the get request failing, but it seems like it should, right?